src/js/modules/comments/actions.js

```diff
@@ -0,0 +1,68 @@
+import * as t from "./actionTypes";
+import { STUY_SPEC_API_HEADERS, STUY_SPEC_API_URL } from "../../constants";
+import axios from "axios";
+import { reset } from "redux-form";
+import { validateKey } from "../../utils";
+
+export const createComment = values => {
+  return dispatch => {
+    dispatch({
+      type: t.CREATE_COMMENT_PENDING,
+      payload: values,
+    });
+    axios
+      .post(`${STUY_SPEC_API_URL}/comments`, values, STUY_SPEC_API_HEADERS)
+      .then(response => {
+        dispatch({
+          type: t.CREATE_COMMENT_FULFILLED,
+          payload: response,
+        });
+        // Destroys the inputs in the form createComment
+        dispatch(reset("createComment"));
+      })
+      .catch(err => {
+        dispatch({
+          type: t.CREATE_COMMENT_REJECTED,
+          payload: err,
+        });
+      });
+  };
+};
+
+export const fetchComments = () => {
+  return dispatch => {
+    dispatch({ type: t.FETCH_COMMENTS_PENDING });
+    axios
+      .get(`${STUY_SPEC_API_URL}/comments`, STUY_SPEC_API_HEADERS)
+      .then(response => {
+        validateComments(response.data);
```

Comment:
How about validate, then dispatch the fulfilled?

Reply:
No reason to split them. validateComments will already throw errors.